cmd/context_register.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright © 2020 Lucas Campos <lucas.campos@axoniq.io>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+)
+
+// constants
+const (
+	rolePrimary       = "PRIMARY"
+	roleActiveBackup  = "ACTIVE_BACKUP"
+	rolePassiveBackup = "PASSIVE_BACKUP"
+	roleMessagingOnly = "MESSAGING_ONLY"
+)
+
+var (
+	context                                           string
+	nodes, activeBackup, passiveBackup, messagingOnly []string
+)
+
+type nodeAndRole struct {
+	Node string `json:"node"`
+	Role string `json:"role"`
+}
+
+type contextNode struct {
+	Context string        `json:"contextNew"`
+	Nodes   []string      `json:"nodes"`
+	Roles   []nodeAndRole `json:"roles"`
+}
+
+var contextRegisterCmd = &cobra.Command{
+	Use:     "register",
+	Aliases: []string{"r"},
+	Short:   "Register a contextNew",
+	Long:    `register a contextNew on axonserver`,
+	Run:     createContext,
+}
+
+func init() {
+	contextCmd.AddCommand(contextRegisterCmd)
+
+	contextRegisterCmd.Flags().StringVarP(&context, "contextNew", "c", "", "Name of the contextNew")
+	contextRegisterCmd.Flags().StringSliceVarP(&nodes, "nodes", "n", []string{}, "[Enterprise Edition only] primary member nodes for contextNew")
+	contextRegisterCmd.Flags().StringSliceVarP(&activeBackup, "activeBackup", "a", []string{}, "[Optional - Enterprise Edition only] active backup member nodes for contextNew")
+	contextRegisterCmd.Flags().StringSliceVarP(&passiveBackup, "passiveBackup", "p", []string{}, "[Optional - Enterprise Edition only] passive backup member nodes for contextNew")
+	contextRegisterCmd.Flags().StringSliceVarP(&messagingOnly, "messagingOnly", "m", []string{}, "[Optional - Enterprise Edition only] messaging-only member nodes for contextNew")
+}
+
+func createContext(cmd *cobra.Command, args []string) {
+	log.Println("calling: " + viper.GetString("server") + contextRegisterURL)
```

Comment:
I see `viper.GetString("server") + aUrl` a lot. Maybe we could extract it to a `const` and use it like that?

Reply:
Let's do it together with the http refactor.